noxfile.py

```diff
@@ -108,8 +113,46 @@ def lint(session: Any) -> None:
     session.run("mypy", "--strict", "test_opensearchpy/test_types/sync_types.py")
 
 
+def lint_per_folder(session: Any) -> None:
+    """
+    allows configuration of pylint rules per folder and runs a pylint command for each folder
+    :param session: the current nox session
+    """
+    # tests should not require function docstrings - tests function names describe themselves;
+    # opensearchpy is generated; may require in the generator code some places
+    default_enable = [
+        "line-too-long",
+        "invalid-name",
+        "pointless-statement",
+        "unspecified-encoding",
+        "missing-function-docstring",
+        "unspecified-encoding",
+    ]
+    override_enable = {
+        "test_opensearchpy/": [
+            "line-too-long",
+            # "invalid-name", lots of short functions with one or two character names
+            "pointless-statement",
+            "unspecified-encoding",
+            "missing-function-docstring",
+            "unspecified-encoding",
+        ],
+        # "opensearchpy/": [""],
+    }
+
+    for source_file in SOURCE_FILES:
+        args = ["--disable=all"]
+        if source_file in override_enable:
+            args.append(f"--enable={','.join(override_enable[source_file])}")
+        else:
+            args.append(f"--enable={','.join(default_enable)}")
+        args.append(source_file)
+        session.run("pylint", *args)
```

Comment:
Unless it's trivial this doesn't need to be part of this PR, but maybe there's a more generic solution where a .pylint file is placed into each folder with the configuration for that folder? This code could iterate over the folders and run it with the contents of that config file.

Reply:
that could be a good solution. I like having all of this in one place, but if the .pylint solution is more standard, then maybe. I'd prefer to get this change in and then investigate.

Reply:
created a new issue: https://github.com/opensearch-project/opensearch-py/issues/645